Only close the span after all writing has finished so that it can sti…

bcluap opened the pull request:
Addresses one part of #16106

Commit:
Only close the span after all writing has finished so that it can still be used
when library interceptors are running

Commit:
Only close the span after all writing has finished

geoand commented on extensions/smallrye-opentracing/runtime/pom.xml:
```diff
@@ -110,6 +110,19 @@
                 </configuration>
             </plugin>
+            <!-- The filter classes needs to be indexed -->
+            <plugin>
```
I think we can forgo this by generating a JaxrsProviderBuildItem with the provider you added

bcluap replied:
Any examples of that I can look at? I'm still a newborn when it comes to extensions :-)

geoand replied:
Any of the usages of https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy-common/spi/src/main/java/io/quarkus/resteasy/common/spi/ResteasyJaxrsProviderBuildItem.java should be just fine 🙂

bcluap replied:
Looking at examples I think this should work:
```
} else if (capabilities.isPresent(Capability.RESTEASY_REACTIVE)) {
            customResponseFilters.produce(new CustomContainerResponseFilterBuildItem(
                    QuarkusSmallRyeTracingStandaloneContainerResponseFilter.class.getName()));
            dynamicFeatures.produce(new DynamicFeatureBuildItem(QuarkusSmallRyeTracingDynamicFeature.class.getName()));
// new...
            providers.produce(
                    new ResteasyJaxrsProviderBuildItem(
                            QuarkusSmallRyeTracingStandaloneContainerResponseFilter.class.getName()));
}
```

But alas its not being registered

geoand replied:
Ah darn. Sorry, the build item I added above is for RESTEasy Classic

geoand replied:
https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/spi-deployment/src/main/java/io/quarkus/resteasy/reactive/spi/ContainerRequestFilterBuildItem.java should be the one to uss

bcluap replied:
How about this one:
writerInterceptors.produce(
                    new WriterInterceptorBuildItem.Builder(
                            QuarkusSmallRyeTracingStandaloneContainerResponseFilter.class.getName()).build());

It works btw

geoand replied:
Ah, it's an interceptor! So yeah, that should be the one to use!

geoand reviewed (commented):
Nice work!

I added a comment about the registration of the provider

Commit:
Only close the span after all writing has finished

Commit:
Only close the span after all writing has finished

Commit:
Only close the span after all writing has finished so that it can still be used
when library interceptors are running

geoand reviewed (commented):
Can you please squash the commits?

Thanks

Commit:
Only close the span after all writing has finished

Commit:
Merge branch '#16106' of https://github.com/bcluap/quarkus into #16106

Commit:
Merge branch '#16106' of https://github.com/bcluap/quarkus into #16106

Commit:
Only close the span after all writing has finished

Only close the span after all writing has finished
Merge branch '#16106' of https://github.com/bcluap/quarkus into #16106
